src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -33,22 +31,29 @@
 import org.springframework.data.mapping.context.MappingContext;
 import org.springframework.lang.Nullable;
 
-public class CouchbaseTemplate implements CouchbaseOperations, ApplicationContextAware {
+/*
+ * @author Michael Nitschinger
+ * @author Michael Reiche
+ */
+
+public class CouchbaseTemplate implements CouchbaseOperations, ApplicationContextAware /*, ApplicationEventPublisherAware */ {
```

Comment:
why is this commented out? please remove if leftover

Reply:
It can come out. I was confused that something that needs to publish application events does not need to be "ApplicationEventPublisherAware".  I suspect that this currently works almost by accident, as CouchbaseTemplate (via CouchbaseTemplateSupport) publishes events using applicationContext.publishEvent(), and not via eventPublisher.

Reply:
removed.